tools/spawn-ants-for-records.mjs

```diff
@@ -0,0 +1,131 @@
+import {
+  ANT,
+  ANT_LUA_ID,
+  AOProcess,
+  ArweaveSigner,
+  createAoSigner,
+  spawnANT,
+} from '@ar.io/sdk';
+import { connect } from '@permaweb/aoconnect';
+import path from 'path';
+import fs from 'fs';
+
+const __dirname = path.dirname(new URL(import.meta.url).pathname);
+const outputFilePath = path.join(__dirname, 'new-arns-processid-mapping.csv');
+
+const wallet = JSON.parse(
+  fs.readFileSync(path.join(__dirname, 'key.json'), 'utf8'),
+);
+const signer = new ArweaveSigner(wallet);
+
+const aoClient = connect({
+  CU_URL: 'https://cu.ar-io.dev',
+});
+
+async function main() {
+  /**
+  Script 1:
+    Inputs: None
+    Outputs: CSV or JSON file with:
+    Domain Name
+    Current ANT Process ID
+Script 2:
+    Inputs: CSV or JSON from Script 1
+    Outputs: CSV or JSON with:
+    Domain name
+    ANT ID of CLONED process
+
+    Primary Business Logic:
+      Pull ANT Info
+      Spawn new ANT
+      Call spawned ANT Initialize-State handler with previous Info
+ */
+
+  // get the csv from the previous script
+  const csv = fs.readFileSync(
+    path.join(__dirname, 'arns-processid-mapping.csv'),
+    'utf8',
+  );
+  const [oldRecordsHeader, ...oldRecords] = csv
+    .split('\n')
+    .map((line) => line.split(','))
+    .filter(([domain, processId]) => domain !== '');
+
+  // create output csv if not exists
+  if (!fs.existsSync(outputFilePath)) {
+    fs.writeFileSync(outputFilePath, 'domain,processId\n', { flag: 'w' });
+  }
+  const [provisionedHeader, ...provisionedRecords] = fs
+    .readFileSync(outputFilePath, 'utf8')
+    .split('\n')
+    .map((line) => line.split(','))
+    .filter(([domain, processId]) => domain !== '');
+
+  const provisionedDomains = provisionedRecords.map(
+    ([domain, processId]) => domain,
+  );
+  // filter out domains that already have an ANT provisioned
+  const recordsToProvision = oldRecords.filter(([domain, processId]) => {
+    return !provisionedDomains.includes(domain);
+  });
+
+  const total = oldRecords.length;
```

Comment:
start at the last name in the provisioned domains file so we can easily restart/continue from where we left off